src/components/NumericStepper/NumericStepper.jsx

```diff
@@ -0,0 +1,330 @@
+	}
+
+	decrease = () => {
+		this.step(-1);
+	}
+
+	componentWillReceiveProps(newProps) {
+		this.setValue(newProps.value);
+	}
+
+	componentWillUpdate(newProps, newState) {
+		if (newState.isCorrected) {
+			const timout = setTimeout(() => {
+				this.setState({
+					isCorrected: !newState.isCorrected
+				});
+				clearTimeout(timout);
+			}, 750);
+		}
+	}
+
+	render() {
+		const {value, isFocused, isCorrected} = this.state;
+
+		const {
+			theme,
+			formatter,
+			min,
+			max,
+			upIconName,
+			downIconName,
+			isDisabled,
+			pattern,
+			InputComponent: Input,
+			ButtonComponent: Button,
+		} = this.props;
+
+		const formatValue = formatter ? formatter(value) : value;
+
+		const inputTheme = {
+			container: classnames(theme.container__input, {
+				[theme.container__input_isCorrected]: isCorrected
+			})
+		};
+
+		const {onWheel, onChange, onBlur, onFocus, onKeyDown} = this;
+
+		const inputProps = {
+			value: formatValue,
+			isDisabled,
+			onBlur,
+			onChange,
+			onFocus,
+			pattern,
+			onKeyDown,
+			theme: inputTheme
+		};
+
+		if (!isDisabled && isFocused) {
+			inputProps.onWheel = onWheel;
+		}
+		const buttonTheme = Object.keys(BUTTON_TYPE).reduce((acc, current) => {
+			return {
+				...acc,
+				[current]: {
+					container: classnames(theme.container__button, theme[`container__button_${BUTTON_TYPE[current]}`]),
+					icon: classnames(theme.container__button__icon)
+				}
+			};
+		}, {});
+
+		return (
+			<div className={theme.container}>
+				<Input {...inputProps} />
+				<div className={theme.container__buttons}>
+					<Holdable onHold={this.onButtonDownClick}
+							  isDisabled={isDisabled || value <= min}>
+						<Button onClick={this.onButtonDownClick}
+								name={downIconName}
+								theme={buttonTheme.UP} />
+					</Holdable>
+					<Holdable onHold={this.onButtonUpClick} isDisabled={isDisabled || value >= max}>
+						<Button onClick={this.onButtonUpClick}
+								theme={buttonTheme.DOWN}
+								name={upIconName} />
+					</Holdable>
+				</div>
+			</div>
+		);
+	}
+
+	onKeyDown = event => {
+		switch (event.keyCode) {
+			case KEYCODE.UP:
+				this.increase();
+				break;
+			case KEYCODE.DOWN:
+				this.decrease();
+				break;
+		}
+	}
+
+	onFocus = () => {
+		this.setState({
+			isFocused: true
+		});
+	}
+
+	onBlur = () => {
+		const {value} = this.state;
+		const {onBlur} = this.props;
+		this.setValue(value);
+		this.setState({
+			isFocused: false
+		});
+
+		onBlur && onBlur();
+	}
+
+	onWheel = e => {
+		if (e.deltaY < 0) {
+			this.increase();
+		} else {
+			this.decrease();
+		}
+	}
+
+	@DEBOUNCE(200)
+	onChange = (value) => {
+		const newValue = parseFloat(value);
+		const {isFocused} = this.state;
+		if (!isFocused) {
+			this.setValue(newValue);
+		} else {
+			this.setState({
+				value: newValue
+			});
+		}
+	}
+
+	onButtonDownClick = () => {
+		this.decrease();
+	}
+
+	onButtonUpClick = () => {
+		this.increase();
+	}
+}
+
+class Holdable extends React.Component {
```

Comment:
Try to `mousedown`, then drag without release and then `mouseup`. Incrementation/Decrementation is cycled.